services/terminal.js

```diff
@@ -46,13 +46,14 @@ class Terminal {
   }
 }
 
-function init(frontend, env, cols, rows) {
+async function init(frontend, env, cols, rows) {
   try {
     const pty = require('node-pty');
     return rpc.handle(new Terminal(frontend, pty, env, cols, rows));
   } catch (e) {
-    frontend.initFailed(e.stack);
-    process.exit(0);
+    await frontend.initFailed(e.stack);
+    Promise.resolve().then(() => process.exit(0));
```

Comment:
Bit puzzled with this code. Is this an elaborate way to do a microtask? Why not just use `setTimeout`?

Reply:
replaced with setImmediate.